trail/upload/process_uploads/header_standardizer.py

```diff
@@ -0,0 +1,329 @@
+    \**kwargs : `dict`
+       Additional keyword arguments
+
+    Keyword arguments
+    -----------------
+    filename : `str`
+        Name of the file from which the HDU was read from, sometimes can encode
+        additional metadata.
+    """
+
+    standardizers = dict()
+    """All registered header standardizers."""
+
+    name = None
+    """Standardizer's name. Only named standardizers will be registered."""
+
+    priority = 0
+    """Priority. Standardizers with high priority are prefered over
+    standardizers with low priority when processing header metadata.
+    """
+
+    @abstractmethod
+    def __init__(self, header, **kwargs):
+        self.header = header
+        self._kwargs = kwargs
+
+    def __init_subclass__(cls, **kwargs):
+        name = getattr(cls, "name", False)
+        if name and name is not None:
+            super().__init_subclass__(**kwargs)
+            HeaderStandardizer.standardizers[cls.name] = cls
+
+    @staticmethod
+    def _computeStandardizedWcs(header, dimX, dimY):
+        """Given an Header containing WCS data and the dimensions of an image
+        calculates the values of world coordinates at image corner and image
+        center and projects them to a unit sphere in Cartesian coordinate
+        system.
+
+        Parameters
+        ----------
+        header : `object`
+            The header, Astropy HDU and its derivatives.
+        dimX : `int`
+            Image dimension in x-axis.
+        dimY : `int`
+            Image dimension in y-axis.
+
+        Returns
+        -------
+        standardizedWcs : `dict`
+            Calculated coorinate values, a dict with wcs_radius,
+            wcs_center_[x, y, z] and wcs_corner_[x, y, z]
+
+        Notes
+        -----
+        The center point is the center of the image as determined its
+        dimensions directly or via header keywords NAXIS1 and NAXIS2, and the
+        corner is taken to be the (0,0)-th pixel.
+        World coords. at the points are to unit sphere, in Cartesian system,
+        and the distance between them is calculated.
+        """
+        standardizedWcs = {}
+        centerX, centerY = int(dimX/2), int(dimY/2)
+
+        # TODO: When eventually logging is added to processing, redirect these
+        # warnings to the logs instead of silencing
+        # NOTE: test if a header doesn't actually have a valid WCS
+        # what is the error raised
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=FITSFixedWarning)
+            wcs = WCS(header)
+
+        centerSkyCoord = wcs.pixel_to_world(centerX, centerY)
+        centerRa = centerSkyCoord.ra.to(u.deg)
+        centerDec = centerSkyCoord.dec.to(u.deg)
+
+        cornerSkyCoord = wcs.pixel_to_world(0, 0)
+        cornerRa = cornerSkyCoord.ra.to(u.deg)
+        cornerDec = cornerSkyCoord.dec.to(u.deg)
+
+        unitSphereCenter = np.array([
+            np.cos(centerDec) * np.cos(centerRa),
```

Comment:
You've forced all the Ras and Decs to be in degrees, and then done trigonometry. Are you 100% sure that's right?

Reply:
Nice catch, thank god for Astropy though because fixing this otherwise would require fixing the test data values too which would be a pain.

```
from astropy import units as u
from astropy.coordinates import SkyCoord
c = SkyCoord(ra=10.625*u.degree, dec=41.2*u.degree, frame='icrs')

c.ra
# <Longitude 10.625 deg>
c.ra.to(u.deg)
# <Longitude 10.625 deg>

import numpy as np
np.cos(c.ra)
# <Quantity 0.98285499>
np.cos(c.ra.to(u.deg))
# <Quantity 0.98285499>
np.cos(c.ra.to(u.rad))
# <Quantity 0.98285499>
```